share/dkg/pedersen/dkg_test.go

```diff
@@ -492,6 +491,56 @@ func TestDKGProcessDeal(t *testing.T) {
 //assert.True(t, dkg2.Finished())
 //}
 
+func TestSetTimeout(t *testing.T) {
+	dkgs = dkgGen()
+	// full secret sharing exchange
+	// 1. broadcast deals
+	resps := make([]*Response, 0, nbParticipants*nbParticipants)
+	for _, dkg := range dkgs {
+		deals, err := dkg.Deals()
+		require.Nil(t, err)
+		for i, d := range deals {
+			resp, err := dkgs[i].ProcessDeal(d)
+			require.Nil(t, err)
+			require.Equal(t, vss.StatusApproval, resp.Response.Status)
+			resps = append(resps, resp)
+		}
+	}
+
+	// 2. Broadcast responses
+	for _, resp := range resps {
+		for _, dkg := range dkgs {
+			if !dkg.verifiers[resp.Index].EnoughApprovals() {
+				// ignore messages about ourself
+				if resp.Response.Index == dkg.index {
+					continue
+				}
+				j, err := dkg.ProcessResponse(resp)
+				require.Nil(t, err)
+				require.Nil(t, j)
+			}
+		}
+	}
+
+	// 3. make sure everyone has the same QUAL set
+	for _, dkg := range dkgs {
+		for _, dkg2 := range dkgs {
+			require.False(t, dkg.isInQUAL(dkg2.index))
+		}
+	}
+
+	for _, dkg := range dkgs {
+		dkg.SetTimeout()
```

Comment:
This should not be needed because all responses are supposed to be dispatch. So you must first make this test fail by not `ProcessResponse` one response on one node, check that it fails, then call `SetTimeout` and check that it succeeds after.

Reply:
I think that is what I tried to simulate? in line 513 I only let nodes processResponses until they have the strict minimum approvals (>= t). I then check that not all DKGs are in all other DKGs's qualifying set, and then I call SetTimeout and check that the qualifying sets are complete this time. Is that incorrect?

Reply:
Oups, you're definitely right, I glossed over the test and thought it was the same as before. My bad!